src/mono/mono/metadata/icall.c

```diff
@@ -8374,6 +8374,115 @@ ves_icall_System_Diagnostics_Debugger_Log (int level, MonoString *volatile* cate
 		mono_get_runtime_callbacks ()->debug_log (level, *category, *message);
 }
 
+#ifdef ENABLE_NETCORE
+#define EVENT_PIPE_DUMMY_PROVIDER_ID 1
```

Comment:
All this stuff (defines and struct definitions) should be in a new `-internal.h` header file.

Also you don't have to put icall definitions in `icall.c`, they could be in a subsystem specific `.c` file.

Reply:
Yes, did it this way short term until I start to make implementation of the icalls in runtime where I will invent a new set of files backing up the complete native implementation of EventPipe.